targets/TARGET_ublox/TARGET_R5030/gpio_api.c

```diff
@@ -18,64 +18,183 @@
+
+	MBED_ASSERT(pin != (PinName)NC);
+
+	obj->reg_base = gpio_base;
 
-    obj->pin = pin;
-    obj->mask = (1ul << pin);
+	obj->pin = pin;
 
-    // Update it when gpio driver is ready
-    //obj->reg_set = &GPIO_OUT_BITSET;
-    //obj->reg_clr = &GPIO_OUT_BITCLR;
-    //obj->reg_val = &GPIO_VALUE;
-   // obj->reg_dir = &GPIO_DIR;
+	pioChannel = gpio_channel_select(obj->pin);
+	obj->mask = (1 << (pioChannel & PIO_CHANNEL_SUB_32_MASK));
 
-    /* Claim the pin */
-    pin_function(pin, PIN_FUNCTION_GPIO);
+	status = gpio_periph_mux_set(PIO_MUX_GPIO,pioChannel,true);
+
+	MBED_ASSERT(status == SUCCESS);
 }
 
+/*****************************************************************************
+								gpio_mode:
+						  Set the mode of gpio
+*****************************************************************************/
 void gpio_mode(gpio_t *obj, PinMode mode)
 {
-    pin_mode(obj->pin, mode);
+    MBED_ASSERT(obj->pin != (PinName)NC);
+
+    switch (mode)
+    {
+      case PullNone:
+    	  obj->reg_base->pio_phdr_0 = obj->mask; // Pull up disable
+    	  obj->reg_base->pio_pldr_0 = obj->mask; // Pull down disable
+    	  break;
+
+      case PullDown:
+    	  obj->reg_base->pio_pler_0 = obj->mask; // Set pull down
+    	  break;
+
+      case PullUp:
+    	  obj->reg_base->pio_pher_0 = obj->mask; // Set pull up
+    	  break;
+
+      default:
+    	  break;
+    }
 }
 
+/*****************************************************************************
+								gpio_dir:
+						Set the direction of gpio
+*****************************************************************************/
 void gpio_dir(gpio_t *obj, PinDirection direction)
 {
     MBED_ASSERT(obj->pin != (PinName)NC);
 
-    switch (direction) {
-        case PIN_INPUT:
-        {
-            *(obj->reg_dir) &= ~(obj->mask);
-        }
-        break;
-        case PIN_OUTPUT:
-        {
-            *(obj->reg_dir) |= obj->mask;
-        }
-        break;
-        /* TODO: looks as though the default is
-         * not normally trapped here */
+    switch (direction)
+    {
+      case PIN_INPUT:
+    	  obj->reg_base->pio_iner_0 = obj->mask; // Enable input
+    	  break;
+
+      case PIN_OUTPUT:
+    	  obj->reg_base->pio_oer_0 = obj->mask; // Enable output
+    	  break;
+    }
+}
+
+/*****************************************************************************
+							pio_periph_muxing_set:
+						    GPIO channel setection
+*****************************************************************************/
+uint8_t gpio_channel_select(PinName pin)
+{
+    switch(pin)
+    {
+       case LED0:
+    	   return GPIO_0_CHANNEL;
+
+       case LED1:
+    	   return GPIO_1_CHANNEL;
+
+       case LED2:
+    	   return GPIO_2_CHANNEL;
+
+       case LED3:
+    	   return GPIO_3_CHANNEL;
+
+       case LED4:
+    	   return GPIO_4_CHANNEL;
+
+       case LED5:
+    	   return GPIO_5_CHANNEL;
+
+       case LED6:
+    	   return GPIO_6_CHANNEL;
+
+       default:
+    	   return ERROR;
```

Comment:
There are 16 GPIOs KM

Reply:
You are returning GPIO_0_CHANNEL and ERROR, both have value 0. May be in default case you want to add print and assign some other value than GPIO_0_CHANNEL